### DISCUSSION TITLE
Change the extensions on kfm/kfm.cmd, and common.cmd

### DISCUSSION THREAD

**Discussion by RealFoobs** [ikemen-engine/Ikemen-GO#947] [Feature Request]
### Is your feature request related to a problem? Please describe.

.cmd is a bad idea for a default extension. Beside sthe fact a complete neophyte will be puzzled when trying to "open" it for the first time, it has some potentially damning side effects:

"Something to mention. The .cmd extension for command files. Not a problem when mugen first came out, everyone used .BAT instead, or .COM. But now .CMD is used as part of the OS and is one of the ways people do limited batch scripting. Because of this, there have been mentions of things like the windows restore stripping .CMD files out of the system when it occurs. Possibly as an anti virus/malware thing. We should be moving to a new extension for these files to A: Make them text rather than something you have to right click to look at and B: To prevent them getting wiped.

As most of the extensions are pretty arbitrary and mugen and FF both look at what the file is within the .def in order to define it, you could simply make up your own standard."

### Describe the solution you'd like

Rename the previously mentioned files to kfm_commands.cns and common_commands.cns. Leave a note in the wiki stating .cmd is a legacy extension that is completely functional, but not recommended, particularly in a full game enviroment where having a single file deleted after restoring windows can render the whole game non-functional.

### Search terms

cmd commands inputs

**Comment by Lazin3ss** [comment]
Well, considering this is a windows-only problem, I think the file deletion issue is mostly reduced to few instances where people might configure things in their computers in an incorrect way, the cmd name resembling an already-known malicious file, or the cmd file really containing malicious code in it. Never in my life I had Windows (7-10) remove .cmd files that weren't viruses.

> there have been mentions of things like the windows restore stripping .CMD files out of the system when it occurs

I need an example of this actually happening. For now, I'm against this idea.

**Comment by RealFoobs** [comment]
Sure thing:

https://mugenguild.com/forum/topics/stop-using-cmd-command-files-143302.0.html

At least 4 people report having dealt with this issue in the past when doing system restoring.

Consider that there is _zero_ benefit in even tempting luck by using this extension.

**Comment by Amidweiz** [comment]
I have noticed that Ikemen GO does care that an zss file has the proper zss extension on files unlike cns where you can have any extension you want and mugen/ikemen go will read it. In turn this means it only really effects cns usage. Personally I don't see a point trying to force a pattern interrupt, for a file extension because windows uses it for batch scripting (even though you can still use .bat on modern windows to do the same). If people want to choose a different file extension they can do so on their own much like that decade old thread brought up.

**Comment by RealFoobs** [comment]
That's the thing, the people that know _know_ because they've been in the game for a long, long time. The main ikemen releases have a broader reach than a decrepit forum thread.  A newcomer isn't going to to know the detriments of using cmd, if they even know how to open it in the first place. 

To complicate matters even further constant.cmd is a new file that has no mugen equivalent to begin with, and the potential to completely cripple an entire installation if its deleted. There is no reason at all for keeping such an important file with such a faulty extension.
